waku/nwaku.go

```diff
@@ -424,6 +424,13 @@ func (w *Waku) DialPeer(address multiaddr.Multiaddr) error {
 	return w.node.Connect(ctx, address)
 }
 
+// TODO: change pubsub topic to shard notation everywhere
+func (w *Waku) RelayPublish(message *pb.WakuMessage, pubsubTopic string) (pb.MessageHash, error) {
+	ctx, cancel := context.WithTimeout(w.ctx, requestTimeout)
+	defer cancel()
+	return w.node.RelayPublish(ctx, message, pubsubTopic)
+}
+
```

Comment:
Here i think we no longer need the `requestTimeout`, as we can receive a context directly in the function and pass it down (the timeout will be extracted anyway from the ctx),  This is so whoever calls RelayPublish is the one determining the timeout of the context, instead of having the bindings decide it for us.

 i.e. something like this:
```suggestion
func (w *Waku) RelayPublish(ctx context.Context, message *pb.WakuMessage, pubsubTopic string) (pb.MessageHash, error) {
	return w.node.RelayPublish(ctx, message, pubsubTopic)
}
```

Reply:
Makes sense!

The pattern that we've been following (at least for the dial functions), is for the functions on the `Waku` struct to not take the context and set it up automatically for the user (so they pass one less parameter), and if the same operation is performed on the `Waku.node` field, then it requires the ctx, and the timeout can be customizable by the user.

Does it make sense? What do you think?

The user should be able to 100% be able to customize the timeout if they want, but I think there should also be some sort of way to publish with a default timeout of ours in case they don't want to set that up themselves every time (or if they don't have idea of what a reasonable timeout is)

Reply:
Not sure tbh. 

In Go i have seen that when doing requests of any type, it's expected to pass a context, and the requester is the one that decides to set up the timeout for the request. If we setup a default request timeout, could a dev still do a request that does not timeout? (i.e. by using a ctx.Background())

There's always the option of exposing a variable like `const RequestTTL = time.Second * 10`, which the dev could then use for their ctx:
```go
ctx, cancel := context.WithTimeout(context.Background(), bindings.RequestTTL)
```

Reply:
Got it! So if it's Go convention to add user-provided ctx in the requests then let's follow it :)

Because we have two functions that do the same, one from the `Waku` object and the other from the `WakuNode` I thought that the higher level (`Waku`) could abstract that away but still making it customizable by the user if they interact directly with the lower level `WakuNode` construct.

However, if it's expected in Go from a user to always provide the ctx, then easier for us :)